src/frequenz/channels/util/_file_watcher.py

```diff
@@ -14,7 +17,17 @@
 from .._exceptions import ReceiverStoppedError
 
 
-class FileWatcher(Receiver[pathlib.Path]):
+@dataclass(frozen=True)
+class FileEvent:
+    """A class to represent a file event useful for notifying file changes."""
+
+    type: Change
+    """The type of change observed."""
+    path: pathlib.Path
+    """The path where the change was observed."""
```

Comment:
Any reason not to put `FileEvent` inside `FileWatcher`, just like `EventType`?

IIRC `EventType` was in the global scope too before, and I think it was changed as part of a restructuring to add the `util` package. I think I did this without thinking too much about google-style imports, and with that in mind it might not be the best approach. It is something we might need to review in the future. But for now I would try to keep the consistency.

Either both inside `FileWatcher` or both outside. About names, if it's inside, I think just plain `Event` is fine, because it must always be referred to by `FileWatcher.Event`. Even more, now the type could be nested in event: `FileWatcher.Event.Type`.

If it goes outside, I think `FileEvent` sounds good, then I would make `EventType` also global for consistency and call it `FileEventType`.

What I like about nesting is that is less import and the context of what each type is used for is super clear, but the downside is there might be more typing.

Reply:
> Any reason not to put FileEvent inside FileWatcher, just like EventType?

The reason was FileWatcher  is a `Receiver[FileEvent]` (inherits from). I would like to define Event internally if that's possible. I'll need to check it out

Reply:
Actually it works just quoting `FileWatcher.Event` in the class signature as follows:
```python
class FileWatcher(Receiver["FileWatcher.Event"]):
    ...
```

Reply:
Then I'll define `Event` inside `FileWatcher`

Reply:
done. Renamed `FileEvent` to `Event` and moved it inside `FileWatcher`